### ISSUE TITLE
Ingress Controller does not create separate listeners for separate ports

### ISSUE THREAD

**Issue by mfacenet** [kubernetes-sigs/aws-alb-ingress-controller#240]
Looks like it's potentially related to https://github.com/coreos/alb-ingress-controller/issues/222.

Specifically if I set something like:
`kind: Ingress
metadata:
  annotations:
    alb.ingress.kubernetes.io/backend-protocol: HTTP
    alb.ingress.kubernetes.io/certificate-arn: xxxxx
    alb.ingress.kubernetes.io/healthcheck-path: /healthz.php
    alb.ingress.kubernetes.io/listen-ports: '[{"HTTP": 80,"HTTPS": 443}]'
    alb.ingress.kubernetes.io/scheme: internet-facing
    alb.ingress.kubernetes.io/security-groups: xxxx
    alb.ingress.kubernetes.io/subnets: xxx
    kubernetes.io/ingress.class: alb
  labels:
  name: test-ingress
  namespace: test-ingress
spec:
  rules:
  - host: ingress-ui-test
    http:
      paths:
      - backend:
          serviceName: ingress
          servicePort: 8200
        path: /
  - host: ingress-api-test
    http:
      paths:
      - backend:
          serviceName: ingress
          servicePort: 8210
        path: /
`

I will get a ALB with rules for the hosts, however, both rules will point to the same listener, in this case it's the listener for the ingress-ui nodeport.

I would expect this to create two listeners, each pointing to a node port, and for the rules to point to the two respective listeners.

**Comment by joshrosso** [comment]
Hi @mfacenet 

I believe this should be fixed in the version below. Let me know if you have a moment to try it out.

```
quay.io/coreos/alb-ingress-controller:1.0-alpha.5
```

If it doesn't let me know the details on what exactly it does setup in AWS.

**Comment by CitySim** [comment]
Hi,
I have the same problem. I have two rules pointing to same service, but a different port. Only one target group is created and used for both rules in the ALB. 

I have tried `quay.io/coreos/alb-ingress-controller:1.0-alpha.5` and `quay.io/coreos/alb-ingress-controller:1.0-alpha.5.1`

config:
```yaml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: example
  namespace: default
  annotations:
    alb.ingress.kubernetes.io/scheme: internal
spec:
  rules:
  - host: admin.example.com
    http:
      paths:
      - path: /
        backend:
          serviceName: coldfusion
          servicePort: 3000
  - http:
      paths:
      - path: /
        backend:
          serviceName: coldfusion
          servicePort: 3001
```

resulting rules:
![alb](https://user-images.githubusercontent.com/109395/31988167-7414c3e8-b96e-11e7-8e83-c42c018bcf44.png)

the port `30842` maps to `coldfusion:3000`

**Comment by onlydole** [comment]
@joshrosso - I am also encountering these errors, even in the latest alpha build too. The error is in the naming of the target group and when the ALB controller goes to look it up. Since it's using the same target name, it gets the zero index result rather than the "right" target group with the correct port.

**Comment by joshrosso** [comment]
Got it, thanks for the extra detail.

How the controller works today is it looks up the NodePort by service name so I can see how this would be an issue.

Shouldn't be a massive change. Will try to tackle it in short order. If anyone has spare cycles and wants to submit a PR, happy to review!

**Comment by bigkraig** [comment]
this should be resolved in later builds, try `1.0-alpha.9`
